### ISSUE TITLE
packaging: Package manual pages in releases

### ISSUE THREAD

**Issue by hseg** [simonmichael/hledger#2309]
(This was flagged in https://github.com/simonmichael/hledger/pull/2223#issuecomment-2352000867, but the reference to the issue being merged into #986 made me think it was being tracked)

Just like other end-user conveniences like command-line completions, I'd expect the manual pages to appear in the binary releases, especially given they've already been written. This should be as simple as adding the various `hledger*/hledger*.1` files to the `tar` invocations in the `binaries-*.yml` workflows.

**Comment by hseg** [comment]
In trying to write a fix for this, noticed daa3010c3e6101fadf8ce6dbcda405cdf94eee06, 69da3c0a17f76eaa5b4508bff3fa4d58c03ab268, etc.
I missed this because I was looking at the 1.41 release.
Am closing this as a duplicate of #986 which I guess has been pushed off for the next milestone instead, please reopen if tracking this separately makes sense.

**Comment by simonmichael** [comment]
Thanks for following up. I’m confused, are manuals included in the 1.41 binaries ? They should be..

**Comment by hseg** [comment]
Nope - https://github.com/simonmichael/hledger/releases/download/1.41/hledger-linux-x64.zip contains the following

[Imagen]

30 dic 2024 20:49:12 Simon Michael ***@***.***>:

**Comment by hseg** [comment]
BTW, any reason why 1.41-branch isn't merged into master?

**Comment by simonmichael** [comment]
It's useful to keep each release branch separate, in case bugfix releases are needed after master has moved on. Any relevant improvements in the release branch will be cherry-picked back to master also.

**Comment by hseg** [comment]
Right, figured that out reading docs/dev later, but I forgot GitHub doesn't send followup emails when posts are edited. The one remaining question (asking again for visibility) is why there is no 1.41.99 tag? Surely every commit to master bumping .version should be tagged A.B.99?

31 dic 2024 18:29:00 Simon Michael ***@***.***>:

**Comment by simonmichael** [comment]
Eh.. yes it exists but maybe I have not pushed that tag. Will check !

**Comment by simonmichael** [comment]
Pushed 1.41.99 to github.

**Comment by hseg** [comment]
I see I missed that this ticket was closed without the underlying issue being
solved. To reiterate -- the 1.41 release does not contain the manual pages, only
the binaries and the bash completion. At the moment, I'm working around this by
manually downloading the manual pages at 1.41 off Github, but it'd be nicer to
just have the package download one tarball and unpack it to the correct
locations.

(What may have been confusing is that I went off on a tangent considering
perhaps packaging master instead, and asking about the branching standard)

**Comment by simonmichael** [comment]
Current status: 
- all the binaries-* workflows in master add the manuals to bindists
- the 1.41 bindists on github have not been retroactively updated, and don't contain the manuals
- 1.42 is imminent, and those bindists will contain manuals

**Comment by hseg** [comment]
Agreed on the decision not to update the 1.41 bindist retroactively, if anything that'd be a point release, and given 1.42 is imminent I doubt it's worth it.

Looking at the referenced commit, I notice you make the stylistic change of replacing `tar -C DIR` by `cd DIR; tar`. Note that this moves the location of the tar archive from `./whatever.tar` to `./DIR/whatever.tar`.
In testing this, though, I notice the original invocation wouldn't've worked, as it mixes traditional and unix-style invocation of `tar`. In particular, the `cvf` should appear as the *first* argument to `tar` in traditional style, or prefixed by `-` in unix style. (One thing this dive has taught me -- in traditional style, the first argument to `tar`, like `printf`'s first argument, determines the arity of the rest of the invocation. For example, `tar cfv etc.tar /etc` is perfectly valid, but `tar -cfv etc.tar /etc` does something completely different)

Finally, I'm ambivalent re shipping the manpages on windows, but given these are easy to add back I can understand the decision.

**Comment by hseg** [comment]
A good overview of the invocation styles is https://www.gnu.org/software/tar/manual/html_section/Styles.html

**Comment by simonmichael** [comment]
Exactly; I had tried to add -C, but discovered it's a nightmare (and varies by tar version) so I reverted to the old behaviour, cd ing into tmp before creating the tarball there.

I'd be happy to consistently include the man pages and shell completions in the Windows bindist, but right now I think 99% of people downloading the windows binary will have no use for those files and they'll just cause clutter/confusion.

**Comment by hseg** [comment]
Fair enough re windows. For my edification, is `-C` that unportable? Or is it just that using it as the first argument means you shouldn't pass `cvf` without a preceding dash, but that's a gotcha that people won't realize?

6 mar 2025 19:59:00 Simon Michael ***@***.***>:

**Comment by simonmichael** [comment]
@hseg, it seemed to me both non portable - in my testing, one of my tar variants (mac and linux) created the tarball in the directory specified by -C and one did not - and too confusing/error-prone to achieve confidence in the command line syntax in that environment.

**Comment by simonmichael** [comment]
I think this can be closed.

**Comment by the-solipsist** [comment]
Given https://github.com/simonmichael/hledger/commit/20c612d0b280fedfdae87d9c5dc1394bd6c8d822 I'm closing this.  If there are still problems (e.g., discovered by the Arch [AUR](https://aur.archlinux.org/pkgbase/hledger-bin#comment-1012829) maintainer) then it can be re-opened.
